src/pages/calendar/components/Section.tsx

```diff
@@ -59,9 +88,36 @@ export function SectionInfo({
   seat,
   meetingTimes,
 }: SectionInfoProps): JSX.Element {
-  const isASYNC = additionalNotes?.indexOf('asynchronous') !== -1;
-  const isSENG = additionalNotes?.indexOf('Reserved for BSENG students') !== -1;
-  const isCSC = additionalNotes?.indexOf('Reserved for students in a Computer Science program') !== -1;
+  const badges: BadgeProps[] = [];
+
+  // Special cases for async/sync/blended because UVic notes are funky
+  if (instructionalMethod === 'online') {
+    if (additionalNotes?.indexOf('mix of “real-time” and asynchronous') !== -1) {
+      badges.push({
+        name: 'Blended',
+        color: 'cyan',
+      });
+    } else if (additionalNotes?.indexOf('fully online and asynchronous') !== -1) {
+      badges.push({
+        name: 'Asynchronous',
+        color: 'cyan',
+      });
+    } else {
+      badges.push({
+        name: 'Synchronous',
+        color: 'cyan',
+      });
+    }
+  }
+
+  for (let i = 0; i < ruleNames.length; i++) {
+    if (additionalNotes?.indexOf(ruleNames[i][0]) !== -1) {
+      badges.push({
+        name: ruleNames[i][1],
+        color: 'red',
+      });
+    }
+  }
```

Comment:
Ideally this is more concise. If you can provide details why its "funky" it would help in the future.